src/Library/Dashboard/SortWidgets.php

```diff
@@ -0,0 +1,196 @@
+
+		$widgetConfigs = $this->getConfigs();
+
+		// First, delete all of our widgets.
+		foreach ( $userWidgetOrder as $container => $widgets ) {
+			$widgets = explode( ',', $widgets );
+
+			foreach ( array_keys( $widgetConfigs ) as $id ) {
+				if ( ( $key = array_search( $id, $widgets ) ) !== false ) {
+					unset( $widgets[ $key ] );
+				}
+			}
+
+			$userWidgetOrder[ $container ] = implode( ',', $widgets );
+		}
+
+		// Then, add them back in.
+		foreach ( $widgetConfigs as $id => $configs ) {
+			switch( empty( $userWidgetOrder[ $configs['container'] ] ) ) {
+				case true:
+					$userWidgetOrder[ $configs['container'] ] = $id;
+					break;
+				case false:
+					$userWidgetOrder[ $configs['container'] ] = $id . ',' . $userWidgetOrder[ $configs['container'] ];
+					break;
+			}
+		}
+		update_user_meta( $userId, 'meta-box-order_dashboard', $userWidgetOrder );
+	}
+
+	/**
+	 * Sort the widgets as containted in the global $wp_meta_boxes.
+	 *
+	 * @since 2.9.0
+	 */
+	public function sortGlobal() {
+		global $wp_meta_boxes;
+
+		foreach ( $this->getConfigs() as $id => $configs ) {
+			$widget = $this->getWidget( $id );
+
+			$invalidData = empty( $widget['container'] ) || empty( $widget['priority'] );
+			if ( $invalidData  ) {
+				continue;
+			}
+
+			// First, remove the widget from where it is now.
+			unset( $wp_meta_boxes['dashboard'][$widget['container']][$widget['priority']][$id] );
+
+			// Then, add it to the correct location.
+			$wp_meta_boxes['dashboard'][ $configs['container'] ][ $configs['priority'] ] =
+				array( $id => $widget['widget'] ) +
+				$wp_meta_boxes['dashboard'][ $configs['container'] ][ $configs['priority'] ];
+		}
+	}
+
+	/**
+	 * Filter the display order of widgets on the dashboard.
+	 *
+	 * Primary objective is to place BoldGrid widgets at the top.
+	 *
+	 * If the user has a custom display order (IE they've dragged things around), we'll force our
+	 * widgets to the top only once. This way, we can force our widgets to the top, but if they want
+	 * to move them lower, they are free to do so.
+	 *
+	 * @priority 20
+	 */
+	public function wp_dashboard_setup() {
+		$userId = get_current_user_id();
+
+		// Debug.
+		// delete_user_meta( $userId, 'meta-box-order_dashboard' );
+		// delete_user_meta( $userId, 'bglibDashboardOrder' );
+
+		// Whether the user has dragged widgets around on their dashboard and changed the order.
+		$userChangedOrder = get_user_meta( get_current_user_id(), 'meta-box-order_dashboard', true );
+		$userChangedOrder = ! empty( $userChangedOrder );
+
+		// Whether bglib has already changed the user's own custom dashboard widget order.
+		$hasChangedLibrary = get_user_meta( $userId, 'bglibDashboardOrder', true );
+		$hasChangedLibrary = ! empty( $hasChangedLibrary );
+
+		switch( $userChangedOrder && ! $hasChangedLibrary ) {
+			case true:
+				$this->sortCustomOrder();
+				break;
+			case false:
+				$this->sortGlobal();
+				break;
+		}
```

Comment:
Why not `$userChangedOrder && ! $hasChangedLibrary ? $this->sortCustomOrder() : $this->sortGlobal();` ?

Reply:
![image](https://user-images.githubusercontent.com/5925563/63190037-5432ef00-c033-11e9-8539-73b60f791509.png)

Reply:
updated.